pkg/artifactory/resource_artifactory_ldap_setting.go

```diff
@@ -0,0 +1,286 @@
+	}
+	return diag.Diagnostics{{
+		Severity: diag.Warning,
+		Summary:  "Usage of Undocumented Artifactory API Endpoints",
+		Detail:   "The artifactory_ldap_setting resource uses endpoints that are undocumented and may not work with SaaS environments, or may change without notice.",
+	}}
+}
+
+func resourceLdapSettingsUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	unpackedLdapSetting := unpackLdapSetting(d)
+
+	var constructBody = map[string]map[string]map[string]LdapSetting{}
+	constructBody["security"] = map[string]map[string]LdapSetting{}
+	constructBody["security"]["ldapSettings"] = map[string]LdapSetting{}
+	constructBody["security"]["ldapSettings"][unpackedLdapSetting.Key] = *unpackedLdapSetting
+	content, err := yaml.Marshal(&constructBody)
+
+	if err != nil {
+		return diag.Errorf("failed to marshal ldap settings during Update")
+	}
+
+	err = sendConfigurationPatch(content, m)
+	if err != nil {
+		return diag.Errorf("failed to send PATCH request to Artifactory during Update")
+	}
+
+	// we should only have one ldap setting resource, using same id
+	d.SetId(unpackedLdapSetting.Key)
+	return resourceLdapSettingsRead(ctx, d, m)
+}
+
+func resourceLdapSettingsDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	rClient := m.(*resty.Client)
+	ldapConfigs := &xmlLdapConfig{}
+
+	rsrcLdapSetting := unpackLdapSetting(d)
+
+	response, err := rClient.R().Get("artifactory/api/system/configuration")
+	if err != nil {
+		return diag.Errorf("failed to retrieve data from <base_url>/artifactory/api/system/configuration during Read")
+	}
+
+	err = xml.Unmarshal(response.Body(), &ldapConfigs)
+	if err != nil {
+		return diag.Errorf("failed to xml unmarshal ldap setting during delete operation")
+	}
+
+	var restoreLdapSettings = map[string]map[string]map[string]LdapSetting{}
+	restoreLdapSettings["security"] = map[string]map[string]LdapSetting{}
+	restoreLdapSettings["security"]["ldapSettings"] = map[string]LdapSetting{}
+
+	for _, iterLdapSetting := range ldapConfigs.LdapSettings.LdapSettingArr {
+		if iterLdapSetting.Key != rsrcLdapSetting.Key {
+			restoreLdapSettings["security"]["ldapSettings"][iterLdapSetting.Key] = iterLdapSetting
+		}
+	}
+
+	var clearAllLdapSettingsConfigs = `
+security:
+  ldapSettings: ~
+`
+	err = sendConfigurationPatch([]byte(clearAllLdapSettingsConfigs), m)
+	if err != nil {
+		return diag.Errorf("failed to send PATCH request to Artifactory during Delete for clearing all Ldap Settings")
+	}
+
+	restoreRestOfLdapSettingsConfigs, err := yaml.Marshal(&restoreLdapSettings)
+	if err != nil {
+		return diag.Errorf("failed to marshal ldap settings during Update")
+	}
+
+	err = sendConfigurationPatch([]byte(restoreRestOfLdapSettingsConfigs), m)
+	if err != nil {
+		return diag.Errorf("failed to send PATCH request to Artifactory during restoration of Ldap Settings")
+	}
+	return nil
+}
+
+func unpackLdapSetting(s *schema.ResourceData) *LdapSetting {
```

Comment:
Why a pointer?

Reply:
I this usage in all our code base. Not sure what i need to change to. Please help with my understanding. Thanks.

Reply:
i see this usage at multiple place in our code. not sure why.